blink/vfs.h

```diff
@@ -0,0 +1,428 @@
+#define VfsTcgetpgrp    tcgetpgrp
+#define VfsTcsetpgrp    tcsetpgrp
+#define VfsTcgetsid     tcgetsid
+#define VfsTcsendbreak  tcsendbreak
+#define VfsTcflow       tcflow
+#define VfsTcflush      tcflush
+#define VfsTcdrain      tcdrain
+#define VfsSockatmark   sockatmark
+#define VfsExecve       execve
+#define VfsMmap         mmap
+#define VfsMunmap       munmap
+#define VfsMprotect     mprotect
+#define VfsMsync        msync
+#else
+#define VfsChown        fchownat
+#define VfsAccess       faccessat
+#define VfsStat         fstatat
+#define VfsChdir        chdir
+#define VfsGetcwd       getcwd
+#define VfsMkdir        mkdirat
+#define VfsChmod        fchmodat
+#define VfsReadlink     readlinkat
+#define VfsOpen         openat
+#define VfsSymlink      symlinkat
+#define VfsMkfifo       mkfifoat
+#define VfsUnlink       unlinkat
+#define VfsRename       renameat
+#define VfsLink         linkat
+#define VfsUtime        utimensat
+#define VfsFchown       fchown
+#define VfsFchdir       fchdir
+#define VfsFchmod       fchmod
+#define VfsFutime       futimens
+#define VfsFstat        fstat
+#define VfsFtruncate    ftruncate
+#define VfsClose        close
+#define VfsRead         read
+#define VfsWrite        write
+#define VfsPread        pread
+#define VfsPwrite       pwrite
+#define VfsReadv        readv
+#define VfsWritev       writev
+#define VfsPreadv       preadv
+#define VfsPwritev      pwritev
+#define VfsSeek         lseek
+#define VfsFsync        fsync
+#define VfsFdatasync    fdatasync
+#define VfsFlock        flock
+#define VfsFcntl        fcntl
+#define VfsDup          dup
+#define VfsDup2         dup2
+#define VfsDup3         dup3
+#define VfsPoll         poll
+#define VfsSelect       pselect
+#define VfsOpendir      fdopendir
+#define VfsSeekdir      seekdir
+#define VfsTelldir      telldir
+#define VfsReaddir      readdir
+#define VfsRewinddir    rewinddir
+#define VfsClosedir     closedir
+#define VfsPipe2        pipe2
+#define VfsSocket       socket
+#define VfsSocketpair   socketpair
+#define VfsBind         bind
+#define VfsConnect      connect
+#define VfsAccept       accept
+#define VfsListen       listen
+#define VfsShutdown     shutdown
+#define VfsRecvmsg      recvmsg
+#define VfsSendmsg      sendmsg
+#define VfsGetsockopt   getsockopt
+#define VfsSetsockopt   setsockopt
+#define VfsGetsockname  getsockname
+#define VfsGetpeername  getpeername
+#define VfsIoctl        ioctl
+#define VfsTcgetattr    tcgetattr
+#define VfsTcsetattr    tcsetattr
+#define VfsTcgetpgrp    tcgetpgrp
+#define VfsTcsetpgrp    tcsetpgrp
+#define VfsTcgetsid     tcgetsid
+#define VfsTcsendbreak  tcsendbreak
+#define VfsTcflow       tcflow
+#define VfsTcflush      tcflush
+#define VfsTcdrain      tcdrain
+#define VfsSockatmark sockatmark
+#define VfsExecve       execve
+#define VfsMmap         mmap
+#define VfsMunmap       munmap
+#define VfsMprotect     mprotect
+#define VfsMsync        msync
+#endif
+
+#define VFS_SYSTEM_ROOT_MOUNT "/SystemRoot"
```

Comment:
Can we not hard code this? Shouldn't this be behind the `-C` flag?

Reply:
> I don't enable it by default for obvious reasons, but it should be enabled for development.
I see that enabling this breaks the build, even after cherry-picking 6d72c93:

```
blink/disinst.c: In function ‘DisName’:
blink/disinst.c:121:37: error: array subscript has type ‘char’ [-Werror=char-subscripts]
  121 |     for (np = name; *np && (islower(*np) || isdigit(*np)); ++np) {
      |                                     ^~~
cc -fno-align-functions -fno-common -pthread -fpie -g -O2 -fno-omit-frame-pointer -fno-optimize-sibling-calls -gdwarf-2 -U_FORTIFY_SOURCE -Wall -Werror -Wno-unused-function -isystem third_party/libz -D_FILE_OFFSET_BITS=64 -D_DARWIN_C_SOURCE -D_DEFAULT_SOURCE -D_BSD_SOURCE -D_GNU_SOURCE -iquote.  -c -o o//blink/divmul.o blink/divmul.c
blink/disinst.c:121:53: error: array subscript has type ‘char’ [-Werror=char-subscripts]
  121 |     for (np = name; *np && (islower(*np) || isdigit(*np)); ++np) {
      |                                                     ^~~
```

Should I fix this?

Reply:
The `-C` flag is the equivalent of the `$BLINK_PREFIX` env var in this PR.
I'll edit it to respect `-C` as well.

`/SystemRoot` is a default `hostfs` mount, similar to the default `devfs` mount at `/dev` and `procfs` mount at `/proc`. This should suffice until we implement `fstab` support.

Reply:
I've fixed my code to respect the `-C` flag when mounting `/` as `hostfs`.
The system root `/` is still mounted to `/SystemRoot`, unless you think that I should keep that under a flag too.